pkgs/games/anki/default.nix

```diff
@@ -1,199 +1,261 @@
-  inherit version;
-  format = "other";
-
-  src = fetchurl {
-    urls = [
-      "https://apps.ankiweb.net/downloads/current/${pname}-${version}-source.tgz"
-      # "https://apps.ankiweb.net/downloads/current/${name}-source.tgz"
-      # "http://ankisrs.net/download/mirror/${name}.tgz"
-      # "http://ankisrs.net/download/mirror/archive/${name}.tgz"
+
+  cargoDeps = rustPlatform.fetchCargoTarball {
+    inherit src;
+    hash = "sha256-m/dcDo9RcVePM+bEEANBY0OZo8+NoK5bYmQJVO9OoHE=";
+  };
+
+  # run with 'python -m black' as part of the build process
+  anki-build-python = python39.withPackages (ps: with ps; [
+    pip
+    black
+    mypy-protobuf
+  ]);
+
+  # anki shells out to git to check its revision, and also to update submodules
+  # We don't actually need the submodules, so we stub that out
+  fakeGit = writeShellScriptBin "git" ''
+    case "$*" in
+      "rev-parse --short=8 HEAD")
+        echo ${builtins.substring 0 8 rev}
+      ;;
+      *"submodule update "*)
+        exit 0
+      ;;
+      *)
+        echo "Unrecognized git: $@"
+        exit 1
+      ;;
+    esac
+  '';
+
+  # We don't want to run pip-sync, it does network-io
+  fakePipSync = writeShellScriptBin "pip-sync" ''
+    exit 0
+  '';
+
+  offlineYarn = writeShellScriptBin "yarn" ''
+    [[ "$1" == "install" ]] && exit 0
+    exec ${yarn}/bin/yarn --offline "$@"
+  '';
+
+  pyEnv = buildEnv {
+    name = "anki-pyenv-${version}";
+    paths = with python39.pkgs; [
+      pip
+      fakePipSync
+      anki-build-python
     ];
-    sha256 = sha256-pkg;
+    pathsToLink = [ "/bin" ];
+  };
+
+  yarnOfflineCache = fetchYarnDeps {
+    yarnLock = "${src}/yarn.lock";
+    hash = "sha256-bAtmMGWi5ETIidFFnG3jzJg2mSBnH5ONO2/Lr9A3PpQ=";
   };
 
+  # https://discourse.nixos.org/t/mkyarnpackage-lockfile-has-incorrect-entry/21586/3
+  anki-nodemodules = stdenv.mkDerivation {
+    pname = "anki-nodemodules";
+    inherit version src yarnOfflineCache;
+
+    nativeBuildInputs = [
+      fixup_yarn_lock
+      yarn
+      nodejs-slim
+    ];
+
+    configurePhase = ''
+      export HOME=$NIX_BUILD_TOP
+      yarn config --offline set yarn-offline-mirror $yarnOfflineCache
+      fixup_yarn_lock yarn.lock
+      yarn install --offline --frozen-lockfile --ignore-scripts --no-progress --non-interactive
+      patchShebangs node_modules/
+      yarn run postinstall --offline
+    '';
+
+    installPhase = ''
+      mv node_modules $out
+    '';
+  };
+in
+python39.pkgs.buildPythonApplication {
+  inherit pname version src;
+
   outputs = [ "out" "doc" "man" ];
 
-  propagatedBuildInputs = [
-    pyqtwebengine
-    sqlalchemy
+  patches = [
+    ./gl-fixup.patch
+    ./no-update-check.patch
+    # Upstreamed in https://github.com/ankitects/anki/pull/2446
+    # We can drop these once we update to an anki version that includes them
+    # already
+    ./0001-Don-t-download-nodejs-if-NODE_BINARY-is-set.patch
+    ./0002-Allow-setting-YARN_BINARY-for-the-build-system.patch
```

Comment:
```suggestion
    # Upstreamed in https://github.com/ankitects/anki/pull/2446
    # We can probably drop this with 2.1.61
    (fetchpatch {
      url = "https://github.com/ankitects/anki/commit/19cf37515284578090071d311e950805a4420ab5.patch";
      hash = "";
    })
```

Reply:
Applying that suggestion, I get:

```
applying patch /nix/store/y6ywlz6cxqmz5pvizx1azhnni3jzhbfr-19cf37515284578090071d311e950805a4420ab5.patch
patching file CONTRIBUTORS
Hunk #1 FAILED at 117.
1 out of 1 hunk FAILED -- saving rejects to file CONTRIBUTORS.rej
```

Upstream squashed the PR's commits, so I can't pick those two patch files and not get the CONTRIBUTORS change from an actual upstream commit.
I think the easiest solution is to leave it as it is

Reply:
fetchpatch has an excludes argument that can be used to not apply the CONTRIBUTORS file:

```
## `fetchpatch` {#fetchpatch}

`fetchpatch` works very similarly to `fetchurl` with the same arguments expected. It expects patch files as a source and performs normalization on them before computing the checksum. For example, it will remove comments or other unstable parts that are sometimes added by version control systems and can change over time.

- `relative`: Similar to using `git-diff`'s `--relative` flag, only keep changes inside the specified directory, making paths relative to it.
- `stripLen`: Remove the first `stripLen` components of pathnames in the patch.
- `extraPrefix`: Prefix pathnames by this string.
- `excludes`: Exclude files matching these patterns (applies after the above arguments).
- `includes`: Include only files matching these patterns (applies after the above arguments).
- `revert`: Revert the patch.

Note that because the checksum is computed after applying these effects, using or modifying these arguments will have no effect unless the `hash` argument is changed as well.
```

But yes, I think keeping the files is just as good, that really doesn't matter :)